src/librustdoc/clean/inline.rs

```diff
@@ -351,7 +351,8 @@ crate fn build_impl(
             }
 
             if let Some(stab) = tcx.lookup_stability(did) {
-                if stab.level.is_unstable() && stab.feature == sym::rustc_private {
+                if let rustc_attr::Unstable { unstables, .. } = &stab.level
+                  && unstables.iter().any(|u| u.feature == sym::rustc_private) {
```

Comment:
Seems it'd be a good idea to have a `has_feature` helper method on `stab.level`.